scipy/signal/_sigtoolsmodule.c

```diff
@@ -977,29 +963,30 @@ static PyObject *_sigtools_median2d(PyObject *NPY_UNUSED(dummy), PyObject *args)
     a_out = (PyArrayObject *)PyArray_SimpleNew(2, PyArray_DIMS(a_image), typenum);
     if (a_out == NULL) goto fail;
 
-    if (setjmp(MALLOC_FAIL)) {
-	PYERR("Memory allocation error.");
-    }
-    else {
-	switch (typenum) {
-	case NPY_UBYTE:
-	    b_medfilt2((unsigned char *)PyArray_DATA(a_image),
+    switch (typenum) {
+        case NPY_UBYTE:
+            b_medfilt2((unsigned char *)PyArray_DATA(a_image),
                        (unsigned char *)PyArray_DATA(a_out),
-                       Nwin, PyArray_DIMS(a_image));
-	    break;
-	case NPY_FLOAT:
-	    f_medfilt2((float *)PyArray_DATA(a_image),
+                       Nwin, PyArray_DIMS(a_image),
+                       &errnum);
+            break;
+        case NPY_FLOAT:
+            f_medfilt2((float *)PyArray_DATA(a_image),
                        (float *)PyArray_DATA(a_out), Nwin,
-                       PyArray_DIMS(a_image));
-	    break;
-	case NPY_DOUBLE:
-	    d_medfilt2((double *)PyArray_DATA(a_image),
+                       PyArray_DIMS(a_image),
+                       &errnum);
+            break;
+        case NPY_DOUBLE:
+            d_medfilt2((double *)PyArray_DATA(a_image),
                        (double *)PyArray_DATA(a_out), Nwin,
-                       PyArray_DIMS(a_image));
-	    break;
-	default:
-	  PYERR("2D median filter only supports uint8, float32, and float64.");
-	}
+                       PyArray_DIMS(a_image),
+                       &errnum);
+            break;
+        default:
+            PYERR("2D median filter only supports uint8, float32, and float64.");
```

Comment:
Minor nit: this signals an internal error, not a user error; other dtypes go through the generic `median_filter` implementation. 

https://github.com/scipy/scipy/blob/v1.14.0/scipy%2Fsignal%2F_signaltools.py#L1938

Reply:
True. I am just making the minimal changes to fix the problem though - this error message is unchanged from what it was.